tessel/src/lib.rs

```diff
@@ -82,86 +78,146 @@ pub struct PortGroup {
 /// use tessel::Port;
 /// ```
 pub struct Port {
-    // Path of the domain socket.
     socket: Arc<Mutex<PortSocket>>,
-    pins: HashMap<usize, Mutex<()>>,
 }
 
-pub struct Pin<'a> {
-    index: usize,
-    _guard: MutexGuard<'a, ()>,
-    socket: Arc<Mutex<PortSocket>>,
-}
+impl Port {
+    pub fn new(path: &str) -> Port {
+        // Create and return the port struct
+        Port {
+            socket: Arc::new(Mutex::new(PortSocket::new(path))),
```

Comment:
Actually the `ports` function really just makes acquisition happen on one thread— it's up to the user to determine whether they want to mutex the result or not. But I'll note this in #29